### PR TITLE
Migrate to jest

### PR DESCRIPTION
**Pull request by yershalom**
Move all tests from mocha to jest for supporting multiple iOS simulators for e2e tests.

### CODE REVIEW

File: e2e/init.js [hunk 1]
```diff
@@ -1,16 +1,25 @@
 const detox = require('detox');
 const config = require('../package.json').detox;
 const exec = require('shell-utils').exec;
+jest.setTimeout(480000);
```
Comment (guyca) [inline e2e/init.js]: Is such a long timeout really needed?

File: package.json [hunk 2]
```diff
@@ -116,9 +117,8 @@
     }
   },
   "detox": {
-    "test-runner": "mocha",
+    "test-runner": "jest",
     "specs": "e2e",
-    "runner-config": "e2e/mocha.opts",
```
Comment (guyca) [inline package.json]: Delete `e2e/mocha.opts`?

Comment (guyca) [reply package.json]: Can we remove mocha from devDependencies now?

File: e2e/init.js (hunk 1 above)
Comment (yershalom) [reply e2e/init.js]: We can lower this, nothing smart behind this number..

File: package.json (hunk 2 above)
Comment (yershalom) [reply package.json]: I'll remove it.

Comment (yershalom) [reply package.json]: And with jest we use e2e/config.json

**Pull request merged**